client/dashboard/leaderboards/top-customers.js

```diff
@@ -0,0 +1,125 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Component } from '@wordpress/element';
+import { map } from 'lodash';
+
+/**
+ * WooCommerce dependencies
+ */
+import { formatCurrency, getCurrencyFormatDecimal } from '@woocommerce/currency';
+import { getNewPath, getPersistedQuery } from '@woocommerce/navigation';
+import { Link } from '@woocommerce/components';
+import { numberFormat } from '@woocommerce/number';
+
+/**
+ * Internal dependencies
+ */
+import Leaderboard from 'analytics/components/leaderboard';
+
+export class TopCustomers extends Component {
+    constructor( props ) {
+        super( props );
+
+        this.getRowsContent = this.getRowsContent.bind( this );
+        this.getHeadersContent = this.getHeadersContent.bind( this );
+    }
+
+    getHeadersContent() {
+        return [
+            {
+                label: __( 'Customer Name', 'woocommerce-admin' ),
+                key: 'name',
+                required: true,
+                isLeftAligned: true,
+                isSortable: true,
+            },
+            {
+                label: __( 'Orders', 'woocommerce-admin' ),
+                key: 'orders_count',
+                required: true,
+                defaultSort: true,
+                isSortable: false,
+                isNumeric: true,
+            },
+            {
+                label: __( 'Total Spend', 'woocommerce-admin' ),
+                key: 'total_spend',
+                isSortable: false,
+                isNumeric: true,
+            },
+        ];
+    }
+
+    getRowsContent( data ) {
+        const { query } = this.props;
+        const persistedQuery = getPersistedQuery( query );
+        return map( data, row => {
+            //const { amount, coupon_id, extended_info, orders_count } = row;
+            const { id, total_spend, name, orders_count, avg_order_value, user_id } = row;
+            console.log(row);
+            //const { code } = extended_info;
+
+            const customerUrl = getNewPath( persistedQuery, 'analytics/customers', {
+                // filter: 'single_',
+                // coupons: coupon_id,
+                search: 'name',
+            } );
+            const customerLink = (
+                <Link href={ customerUrl } type="wc-admin">
+                { name }
+                </Link>
+        );
+
+            const ordersUrl = getNewPath( persistedQuery, 'analytics/orders', {
```

Comment:
We don't have a customer filter on the orders report, so we might want to remove and unlink this one for now.